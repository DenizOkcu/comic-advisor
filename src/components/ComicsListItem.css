.list-item {
  border-radius: 2px;
  border: 1px solid #ccc;
  box-shadow: 2px 2px 5px #eee;
  box-sizing: border-box;
  cursor: pointer;
  display: flex;
  margin-bottom: 10px;
}

.list-item-cover {
  flex: 1;
  border-radius: 2px;
}

.list-item-description {
  flex: 2;
  display: flex;
  flex-direction: column;
  padding: 3px 3px 0 10px;
}

/* only show active and hover effects in the master/detail view */
@media (min-width: 550px) {
  .list-item.active,
  .list-item:hover {
    background-color: #eee;
  }
}
