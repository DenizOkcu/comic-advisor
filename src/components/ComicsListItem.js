import React from "react";
import { Row } from "./grid";

const ComicsListItem = ({ comic, selectComic }) => {
  const writerSection = comic.writer ? (
    <Row>
      <div className="twelve columns">{comic.writer.name}</div>
    </Row>
  ) : (
    ""
  );
  const clickToSelectComic = () => {
    selectComic(comic.id);
  };

  return (
    <Row>
      <section className="comics-list-item" onClick={clickToSelectComic}>
        <div className="four columns">
          <img src={comic.coverPath} className="thumb" alt={comic.title} />
        </div>
        <div className="eight columns ">
          <Row>
            <div className="twelve columns list-item-title">{comic.title}</div>
          </Row>
          {writerSection}
          <Row>
            <div className="twelve columns">
              Issue: {comic.issueNumber} - {comic.price}
            </div>
          </Row>
        </div>
      </section>
    </Row>
  );
};

export default ComicsListItem;
