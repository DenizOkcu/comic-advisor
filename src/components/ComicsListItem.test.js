import React from "react";
import { shallow } from "enzyme";
import ComicsListItem from "./ComicsListItem";

let comic;
let selectComicMock;

beforeEach(() => {
  comic = {
    id: 1,
    writer: { name: "Bernd", role: "writer" },
    prices: 3.99,
    title: "Superman",
    coverPath: "cover.png",
    pageCount: 32
  };
  selectComicMock = jest.fn();
});

describe("ComicsListItem", () => {
  it("should render correctly a nonactive list item", () => {
    const component = shallow(
      <ComicsListItem
        comic={comic}
        active={false}
        selectComic={selectComicMock}
        debug
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("should render correctly an active list item", () => {
    const component = shallow(
      <ComicsListItem
        comic={comic}
        active={true}
        selectComic={selectComicMock}
        debug
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("should render correctly without writer", () => {
    comic.writer = null;
    const component = shallow(
      <ComicsListItem
        comic={comic}
        active={true}
        selectComic={selectComicMock}
        debug
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("should call the selectComic function on click", () => {
    const component = shallow(
      <ComicsListItem
        comic={comic}
        active={true}
        selectComic={selectComicMock}
        debug
      />
    );

    expect(selectComicMock.mock.calls.length).toEqual(0);
    component.find(".list-item").simulate("click");
    expect(selectComicMock.mock.calls.length).toEqual(1);
    expect(selectComicMock).toBeCalledWith(1);
  });
});
