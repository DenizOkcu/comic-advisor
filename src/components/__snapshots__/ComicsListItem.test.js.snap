// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicsListItem should render correctly in "debug" mode 1`] = `
<div
  className="row"
  onClick={[Function]}
>
  <div
    className="fixed four columns "
  >
    <img
      alt="title"
      className="thumb"
      src="test.jpg"
    />
  </div>
  <div
    className="fixed eight columns "
  >
    <div
      className="row"
    >
      <div
        className="twelve columns list-item-title"
      >
        title
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="twelve columns"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="twelve columns"
      >
        32
         Pages - $
        3.99
      </div>
    </div>
  </div>
</div>
`;
